src/cli/clean.rs

```diff
@@ -0,0 +1,144 @@
+/// Command to clean the parts of your system which are touched by pixi.
+use crate::{config, EnvironmentName, Project};
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::time::Duration;
+
+use crate::progress::{global_multi_progress, long_running_progress_style};
+use clap::Parser;
+use indicatif::ProgressBar;
+
+#[derive(Parser, Debug)]
+pub enum Command {
+    #[clap(name = "cache")]
+    Cache(CacheArgs),
+}
+
+/// Clean the parts of your system which are touched by pixi.
+/// Defaults to cleaning the environments and task cache.
+/// Use the `cache` subcommand to clean the cache.
+#[derive(Parser, Debug)]
+pub struct Args {
+    #[command(subcommand)]
+    command: Option<Command>,
+    /// The path to 'pixi.toml' or 'pyproject.toml'
+    #[arg(long)]
+    pub manifest_path: Option<PathBuf>,
+
+    /// The environment directory to remove.
+    #[arg(long, short, conflicts_with = "command")]
+    pub environment: Option<String>,
+}
+
+/// Clean the cache of your system which are touched by pixi.
+#[derive(Parser, Debug)]
+pub struct CacheArgs {
+    /// Clean only the pypi related cache.
+    #[arg(long)]
+    pub pypi: bool,
+
+    /// Clean only the conda related cache.
+    #[arg(long)]
+    pub conda: bool,
+    // TODO: Would be amazing to have a --unused flag to clean only the unused cache.
```

Comment:
just my 2 cents - we need to be sure that we can support windows with this. I think they don't have inode count